`Tab` documentation update (#1926)
hashicorp/design-system: website/docs/components/tabs/partials/guidelines/guidelines.md

![Tabs with external trigger](/assets/components/tabs/tabs-external-trigger.png)
!!!

> ## Size

Author: ```suggestion
## Sizes
```

> ## Sizes

Reviewer: In general, I think we can expand on this a little bit and be slightly more prescriptive around where we expect the different sizes to be used. A couple of things that I might consider:

- I think you hit the nail on the head with hierarchical distinction, can this come to the forefront a bit? This was the main problem the additional size was aiming to solve.
- Would be good to include something about nesting, since the large variant sits higher in the hierarchy, this could be a more effective way of communicating nested/dependent content; or, if this is something we recommend against (not mixing sizes), why, and some other solutions
- There's already a bit of content around "contained" tabs that might be useful to consider as well, _in general_ it seems like the large version should live more at the page level and the medium version should live in a contained environment a bit better.

Initial thoughts here, happy to pair on this next week to brainstorm some additional content!

> For most scenarios, medium-sized tabs fit various UI contexts well. Choose the size that aligns with the content's complexity and desired interaction pace. Large tabs should emphasize important sections, enhancing clarity and hierarchical distinction.

Reviewer 2: ```suggestion
Medium-sized tabs work best in most contexts. Large tabs can be used when needed to emphasize important sections, enhance clarity, and convey hierarchical distinction.
```

![Tabs with external trigger](/assets/components/tabs/tabs-external-trigger.png)
!!!

## Sizes

> Medium-sized tabs work best in most contexts. Large tabs can be used when needed to emphasize important sections, enhance clarity, and convey hierarchical distinction. 

Reviewer: ```suggestion
Medium-sized tabs work best in most contexts. Large tabs can be used to emphasize important sections, enhance clarity, and convey hierarchical distinction. 
```

I think you can drop "when needed" here

![Tabs sizes example](/assets/components/tabs/tab-size.png)

### Choosing sizes 

> Mixing different tab sizes is not recommended, as it can confuse users. Choose tab sizes with a purpose, aligning them with the hierarchical importance of the information they represent. 

Reviewer: Couple of suggestions here

```suggestion
Mixing different tab sizes is not recommended and can negatively impact the natural page hierarchy. Choose tab sizes intentionally and aligning them with the hierarchical importance of the information they represent. 
```

Reviewer 2: Made an additional small grammar fix in the above suggestion.

"... **aligning** them with the hierarchical importance..."
=>
"... **align** them with the hierarchical importance..."

> ### Organizing 

Reviewer: I almost could see removing this headline entirely as the content in each section is all related to sizing.

Reviewer 2: I would agree with removing this "### Organizing" header. I don't think it adds any useful distinction.

> ### Organizing 

Author: ```suggestion

```

> We recommend to structure your tabs to reflect the hierarchical organization. Large tabs are reserved for major differences, emphasizing their importance, and should be used sparingly. Medium tabs, by contrast, are more adaptable and suited for grouping content under those significant categories, promoting an organized and straightforward experience within specific sections or among related content.

Reviewer: ```suggestion
We recommend structuring Tabs to reflect the intended hierarchy of the content they contain. Large tabs are reserved for major differences, emphasizing their importance, and should be used sparingly. Medium tabs, by contrast, are more adaptable and suited for grouping content under those significant categories, promoting an organized and straightforward experience within specific sections or among related content.
```

It actually might be reasonable to break these into bullets (kind of what we talked about today), maybe consider this:

We recommend structuring Tabs to reflect the intended hierarchy of the content they contain:

- Large Tabs sit higher in the page hierarchy and can be used to express more important content, but should be used more sparingly.
- Medium tabs are more adaptable in most scenarios and naturally occupy the same hierarchy as most page content.

Reviewer 2: I think the suggested refactor using bulleted points is much clearer.

![Tabs with external trigger](/assets/components/tabs/tabs-external-trigger.png)
!!!

## Sizes

> Medium-sized tabs work best in most contexts. Large tabs can be used to emphasize important sections, enhance clarity, and convey hierarchical distinction. 

Reviewer: ```suggestion
Medium tabs are the default and work best in most contexts. Large tabs can be used to emphasize important sections, enhance clarity, and convey hierarchical distinction. 
```

One other note; I think you can remove this for consistency with the rest of the language. Also, per @KristinLBradley 's recommendation, could add in a note about the default here.
